File: LABA5/commandsPack/Info.java
package commandsPack;

import elementsPack.Movie;
import mainPack.MovieCollection;

public class Info extends Command{
    public Info(MovieCollection collection) {
        this.collection = collection;
    }
    @Override
    public void execute() {
        int n = 0;
        System.out.println("Collection type: " + Movie.class +
                "\nDate of initialization: " + collection.getDateOfCreation() +
                "\nNumber of elements stored: " + collection.getNumberOfElements() +
                '\n');
    }
}

File: LABA5kt/src/commandsPack/Info.kt
package commandsPack

import commandInterfacePack.Info
import elementsPack.Movie

/**
 * Class storing method info
 */
open class Info protected constructor() : Help(), Info {
    /**
     * This method prints main info about current collection
     */
    override fun info() {
        println(
            """Collection type: ${Movie::class.java.simpleName}
Date of initialization: ${collection!!.getDateOfCreation()}
Number of elements stored: ${collection!!.numberOfElements}"""
        )
    }
}